azurerm/internal/services/containers/kubernetes_cluster_node_pool_data_source.go

```diff
@@ -140,6 +140,8 @@ func dataSourceKubernetesClusterNodePool() *schema.Resource {
 
 			"tags": tags.SchemaDataSource(),
 
+			"upgrade_settings": upgradeSettingsSchema(),
```

Comment:
since this is a data source this wants to be:

```suggestion
			"upgrade_settings": upgradeSettingsForDataSourceSchema(),
```